rerun-failed-run/rerun-failed-run.yml

```diff
@@ -0,0 +1,73 @@
+name: Retrigger failed run action
+
+on:
+  workflow_dispatch:
+    inputs:
+      run_id:
+        description: The id of the run which we want to retrigger
+        required: true
+      error_message:
+        description: Custom error message to look for in the logs.
+        required: true
+      repository:
+        description: Github repository
+        required: true
+jobs:
+  rerun:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@v2.7.4
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/data/products/infra/ci/retrigger-gha-workflow RETRIGGER_APP_KEY;
+            secret/data/products/infra/ci/retrigger-gha-workflow RETRIGGER_APP_ID;
+
+      - name: Generate a GitHub token
+        id: github-token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ steps.secrets.outputs.RETRIGGER_APP_ID }}
+          private_key: ${{ steps.secrets.outputs.RETRIGGER_APP_KEY }}
+
+      - name: Check previous run logs
+        id: check_logs
+        env:
+          GH_DEBUG: api
+        run: |
+          echo ${{ steps.github-token.outputs.token }} | gh auth login --with-token
+          echo "Fetching logs for run with id: ${{ inputs.run_id }} from ${{ inputs.repository }} repository."
+          # Fetch the logs from the previous run
+          logs=$(gh run view ${{ inputs.run_id }} -R ${{ inputs.repository }} --log)
+          echo "Logs coming from run with id: ${{ inputs.run_id }}"
+          echo "$logs"
+          echo "end logs"
+          # Check if the desired string is present in the logs
+          if echo "$logs" | grep -F "${{ inputs.error_message }}" >/dev/null; then
+            count=$(echo "$logs" | grep -F "${{ inputs.error_message }}" | wc -l)
+            echo "count: $count"
+            if [ "$count" -gt 2 ]; then
```

Comment:
Wny only a single occurence of the message is not sufficient to trigger a return? `"$count" -gt 2` means it will only trigger when there's 3 or more line matches, why is that?

Reply:
This is an example of a run where the log message has actually been encountered in the run:

Logs from the run start [here](https://github.com/camunda/infra-test-retrying-failed-actions/actions/runs/7830046910/job/21363225321#step:4:692) and finish [here](https://github.com/camunda/infra-test-retrying-failed-actions/actions/runs/7830046910/job/21363225321#step:4:1114). In these logs the inputted error message `Process completed with exit code 1.` is being encountered 3 times. Two of these times is because it is being inputted in the workflow [1](https://github.com/camunda/infra-test-retrying-failed-actions/actions/runs/7830046910/job/21363225321#step:4:1081), [2](https://github.com/camunda/infra-test-retrying-failed-actions/actions/runs/7830046910/job/21363225321#step:4:896) and one time because the error message is [actually encountered](https://github.com/camunda/infra-test-retrying-failed-actions/actions/runs/7830046910/job/21363225321#step:4:834). So if the error message is encountered in the logs then it will show up only two times.

Reply:
[This](https://github.com/camunda/infra-test-retrying-failed-actions/actions/runs/7829741834/job/21362332769#step:4:1081) is an example from a workflow where the error message to compare against was `Random error message.`

Reply:
I see. Is there a way of excluding these 2 cases from count? The current implementation is quite brittle, relying that 2 extra mentions should happen in the logs. For example, if one day we stop passing `GH_DEBUG: api`  as an env var, it will be only printed once.

Reply:
This is actually a very valid point. I replaced it with 
`gh run view ${{ inputs.run_id }} -R ${{ inputs.repository }} --exit-status `

That command will show only the exit status of the run of the job and the reason why it fails

Reply:
we can therefore search if there is more than 0 occurancies of the error message